### PR TITLE
Docs: Refactor documentation

### LINKED ISSUE #347: [Doc]: Update the doc format to support adaptative display
### Documentation Location

All the documentation

### Current issue

- The Header title is not positioned correctly
- The current page format is not adaptative
  - no breakpoint header change
  - not compatible with the side navigation used today
- There is no footer  

### Expected Correction

TBD with @julien-deramond

### PR DESCRIPTION
**Pull request by louismaximepiton**
### Description

Certainly forgot to list some things in here but here are the main changes that this PR brings up.

#### First commit

* Update CONTRIBUTING.md to better fit to the project structure
* Refactoring of the main page, removing some CSS as well
* Adding the Orange font inside the doc
* Introducing some partials for header and footer for clearer template

#### Second commit

* :warning: Removed all `/404.html` that were dupes since they are a permalink and were never used because `/404.html` is predominant -> Removes a warning inside the serve console
* Added a page to explain the license and redirect to the license file directly
* Added a page to present the team maintaining the library -> :warning: Should we keep this kind of page ?-> `/data/team.yml` Need to be maintained by core team
* Added an offcanvas wraping the main navigation so it's usable from anywhere inside the page
* Added a default template that was missing for `404.html` page
* Added a title to each `*_docs.md` so it's now displayed as the title of the page
* Wrapped all markdown table with `<div class="table-responsive" markdown="1">...</div>` to make sure the mobile documentation is still readable at any time
* Tweaked a bit `/index*.md` to make sure the workaround inside the main navigation still works

#### Third commit

* Added some comfort updates:
  * No need to update the table of content anymore, just add a section with `* Table of contents\n{:toc}` to have it rendered inside pages
  * No need to add the `{:.table}` before each table anymore (see `/_plugins/table-responsive.rb`). If the plugin is added to `/_plugins` it's auto imported inside the project (modulo dependencies)
  * Tables are now responsive by default with markdown template
* Update the `detail.html` to have the same rendering as the documentation

#### `/_layouts/main.html`

* Kept the CSS part as tidy as possible there are some `style` tags but I feel more comfy with this.
* Reorganized the main html part
* Set the main navigation in an offcanvas on mobile viewsize. It's toggled via the global header burger menu
* Updated the rendering of the global site to have a render closer to what Boosted propose.
* May need some updates while updating Boosted version
* Updated the main navigation to reflect which page is active
* Updated the `<header><main><footer>` places to have a better a11y base

### Related issues

Fixes #347

### CODE REVIEW

**Review by paulinea — APPROVED** [review]

File: docs/_includes/header.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<header class="sticky-top bg-black">
+  <div class="navbar navbar-dark bg-dark navbar-expand-lg" aria-label="Global header of ODS Android">
+    <div class="container-xxl">
+      <!-- Orange brand logo -->
+      <div class="navbar-brand me-auto me-lg-4">
+        <span>
+          <img src="https://c.woopic.com/logo-orange.png" width="50" height="50" alt="ODS Android" loading="lazy">
```
Comment (julien-deramond) [inline docs/_includes/header.html:7]: I think this is not the right icon, it should be the SVG version which contains the minified version at some breakpoints. Given the fact that there won't be too many users related to this documentation, you could try to use this one https://boosted.orange.com/docs/5.3/assets/brand/orange-logo.svg.
I know it's linked to a fixed 5.3 version, but this image is not in the Woopic CDN for now.

**Review by julien-deramond — CHANGES_REQUESTED** [review]
I haven't finished the review for the HTML part, but here are some first comments.

**Review by julien-deramond — APPROVED** [review]
LGTM. Some enhancements could be made to the HTML part but let's do them step-by-step in the future. This version is already a great improvement for the users 🚀 
Thanks for the PR @louismaximepiton

File: docs/modules/About_docs.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ---
 layout: main
+title: About
```
[the remark this answers is not in the document]
Comment (louismaximepiton) [reply docs/modules/About_docs.md:3]: Good catch, it should be resolved by https://github.com/Orange-OpenSource/ods-android/pull/808/commits/a33440625ef3645e34870fed2b8e9e53682348d2

**Review by florentmaitre — APPROVED** [review]

**Pull request merged**